### PR TITLE
Kernel/Storage: Bring NVMe code closer to the repository programming style

### PR DESCRIPTION
**Pull request by supercomputer7**
A bunch of fixes before landing support for polling mode for NVMe devices to allow @tomuta machine to boot (#11594).

### CODE REVIEW

File: Kernel/Storage/NVMe/NVMeQueue.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace Kernel {
 ErrorOr<NonnullRefPtr<NVMeQueue>> NVMeQueue::try_create(u16 qid, u8 irq, u32 q_depth, OwnPtr<Memory::Region> cq_dma_region, NonnullRefPtrVector<Memory::PhysicalPage> cq_dma_page, OwnPtr<Memory::Region> sq_dma_region, NonnullRefPtrVector<Memory::PhysicalPage> sq_dma_page, Memory::TypedMapping<volatile DoorbellRegister> db_regs)
 {
     auto queue = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) NVMeQueue(qid, irq, q_depth, move(cq_dma_region), cq_dma_page, move(sq_dma_region), sq_dma_page, move(db_regs))));
-    TRY(queue->create());
+    TRY(queue->try_create_dma_buffer());
```
Comment (IdanHo) [inline Kernel/Storage/NVMe/NVMeQueue.cpp:19]: Not your fault of course, but would be nice to solve as well: two phase construction is pretty rare in serenity (and rightfully so), so doing this before the constructor is called would be nice.

Comment (supercomputer7) [reply Kernel/Storage/NVMe/NVMeQueue.cpp]: Done :)

File: Kernel/Storage/NVMe/NVMeQueue.cpp [hunk 2]
```diff
@@ -15,12 +15,13 @@ namespace Kernel {
 
 ErrorOr<NonnullRefPtr<NVMeQueue>> NVMeQueue::try_create(u16 qid, u8 irq, u32 q_depth, OwnPtr<Memory::Region> cq_dma_region, NonnullRefPtrVector<Memory::PhysicalPage> cq_dma_page, OwnPtr<Memory::Region> sq_dma_region, NonnullRefPtrVector<Memory::PhysicalPage> sq_dma_page, Memory::TypedMapping<volatile DoorbellRegister> db_regs)
 {
-    auto queue = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) NVMeQueue(qid, irq, q_depth, move(cq_dma_region), cq_dma_page, move(sq_dma_region), sq_dma_page, move(db_regs))));
-    TRY(queue->create());
+    // Note: Allocate DMA region for RW operation. For now the requests don't exceed more than 4096 bytes (Storage device takes care of it)
+    auto dma_region = TRY(MM.allocate_dma_buffer_page("NVMe Queue"sv, Memory::Region::Access::ReadWrite, m_rw_dma_page));
```
Comment (IdanHo) [inline Kernel/Storage/NVMe/NVMeQueue.cpp:19]: `m_rw_dma_page` does this work :thinking:

Comment (supercomputer7) [reply Kernel/Storage/NVMe/NVMeQueue.cpp]: Oops, I think I solved it :)

**Comment by supercomputer7** [comment]
@IdanHo any suggestions on this or we can merge this?

**Pull request merged**